[Lesson 16] Add waiters and implement new test:

TatyanaDovzhich opened the pull request:
- Add implicitly wait on driver level
- Add explicitly waits on base page level
- Add decorator to handle "magic-strange-weird" Sign Up button
- Added test for post creation with related constants and pages

TatyanaDovzhich committed:
[Lesson 16] Add waiters and implement new test:

- Add implicitly wait on driver level
- Add explicitly waits on base page level
- Add decorator to handle "magic-strange-weird" Sign Up button
- Added test for post creation with related constants and pages

TatyanaDovzhich committed:
[Lesson 16_Homework 16] Add waiters:

- Add implicitly wait on driver level (conftest.py)
- Add URLs for CreatPost and Post Pages (base.py)

QALightLecorPython commented on conftest.py:
```diff
@@ -17,4 +20,29 @@ def driver():
+
+
+@pytest.fixture()
+def hello_page(driver):
```
It is not working like this. You cannot just create any page. Only start page can be created from driver by opening it, all other pages are have to be crated by navigating to them under the page objects

TatyanaDovzhich replied:
Fixings are:
-  delete unnecessary wrong drivers at conftest.py;
- create waiter for Sign In button at the Start Page;
- create waiters for Create Post and Sign Out button at the header;
- create waiter for Save New Post button at the Create Post page.

TatyanaDovzhich committed:
[Lesson 16_Homework 16] Fix#01: Delete unnecessary wrong drivers at conftest.py

TatyanaDovzhich committed:
[Lesson 16_Homework 16] Fix#02: Delete unnecessary wrong URLs at constants/base.py

TatyanaDovzhich committed:
[Lesson 16_Homework 16] Fix#03: Some fixes at base_page.py and utils.py

TatyanaDovzhich committed:
[Lesson 16_Homework 16] Fix#04:
- Create constants/header.py and pages/header.py
- Create waiter for Sign In button at the Start Page
- Create waiters for Create Post and Sign Out button at the header
- Create waiter for Save New Post button at the Create Post page

QALightLecorPython commented on pages/create_post_page.py:
```diff
@@ -1,7 +1,9 @@
-import logging
+from enum import verify
```
what is this ?

QALightLecorPython commented on pages/create_post_page.py:
```diff
@@ -10,12 +12,26 @@ class CreatePostPage(BasePage):
         self.fill_field(xpath=self.const.TITLE_INPUT_XPATH, value=title)
         self.fill_field(xpath=self.const.BODY_AREA_XPATH, value=body)
+        # Click on Save Post button
+        if verify:
```
I think you don't need this logic here. But if I wrong, you should add additional argument to the method and pass it from the test

QALightLecorPython commented on pages/create_post_page.py:
```diff
@@ -10,12 +12,26 @@ class CreatePostPage(BasePage):
+        self.header = Header(self.driver)
+
+    @wait_until_ok(timeout=3, period=0.5)
+    def click_and_validate_save_post_button(self):
```
Do you have a trouble with this button or you add this `just in case` ?

QALightLecorPython commented on pages/header.py:
```diff
@@ -0,0 +1,53 @@
+            self.click(self.const.SIGN_OUT_BUTTON_XPATH)
+
+    # @log_wrapper
+    def navigate_to_create_post(self):
```
This method should return CreatePostPage

QALightLecorPython commented on pages/header.py:
```diff
@@ -0,0 +1,53 @@
+        self.const = HeaderConsts
+
+    @wait_until_ok(timeout=3, period=0.5)
+    def click_and_validate_create_post_button(self):
```
Why do you need this verification ?
Anyway this button won't disappear as header is always exists.

QALightLecorPython commented on pages/header.py:
```diff
@@ -0,0 +1,53 @@
+            self.click(self.const.CREATE_POST_BUTTON_XPATH)
+
+    @wait_until_ok(timeout=3, period=0.5)
+    def click_and_validate_sign_out_button(self):
```
Why do you need this ? You already have `navigate_to_start_page`

QALightLecorPython reviewed (changes requested):
I don't get the reason of some changes, please fix or let me know why did you do them )
